src/main/java/com/jw/service/OrderService.java

```diff
@@ -40,10 +49,20 @@ public OrderResponse getOrderById(Long id) {
         return orderMapper.toOrderResponse(order);
     }
 
+    public OrderResponse processUpdateOrder(Long orderId, OrderRequest orderRequest) {
+        checkIfOrderExistsOrElseThrowException(orderId);
+        Order order = orderMapper.toOrder(orderRequest);
+        order.setOrderId(orderId);
+        String status = orderRepository.findById(orderId).getStatus();
+        order.setStatus(status);
+        updateOrder(order);
+        return orderMapper.toOrderResponse(order);
+    }
+
     @Transactional
```

Comment:
As I can see this method is been called by `processUpdateOrder` which is present at the same class, so it's an internal call without it goes through the proxy. So it'll cause that the @Transaction annotation will not work in here, you should move this annotation to `processUpdateOrder` method, so it'll start to work properly